packages/block-editor/src/components/line-height-control/index.js

```diff
@@ -45,7 +45,7 @@ const LineHeightControl = ( {
 			case '0': {
 				// This means the user explicitly input '0', rather than stepped down
 				// from an undefined value state.
-				if ( wasTypedOrPasted ) return nextValue;
+				if ( wasTypedOrPasted ) return parseInt( nextValue, 10 );
```

Comment:
Do you remember why you had to make this specific change?

Reply:
It was added in this [commit](https://github.com/WordPress/gutenberg/pull/48917/commits/3c12a3afdb9eafe9721c1829b5aa482a38a3da02). The line height control was getting inconsistent types back depending on how values were entered. This normalized the next value to a number.

Reply:
If I'm not wrong, if you normalize it to a string instead, the changes to `cleanEmptyObject` won't be needed in this

Reply:
I can make that change.

The other return values around it were numbers and this component is built on the NumberControl, so my understanding was that a numeric value was expected.  I know we have a lot of inconsistencies around `number` vs `string` types for `NumberControl` related components and those are being addressed in https://github.com/WordPress/gutenberg/pull/45982. 

Digging deeper, it seems the values passed to `onChange` vary between strings and numbers a lot based on how the value is edited. For example;
- Typing into the field will pass a string
- If you then press enter after typing a value, onChange is called again this time with a number
- If you press up/down arrow keys, the value will be a number
- If you use the custom spin buttons, the value is a string
- Dragging up or down will pass a number value

Given the point of this PR was to fix the application of zero values, I'll normalize to a string as suggested to limit the scope of change. Any wrangling of the value types passed to the change handlers and state reducers can be a follow-up if there is an appetite for it post #45982.

Reply:
I agree with you that the logical thing to do here is to actually use a number because "line height" is a number. But yeah maybe because of the fact that it surfaces the other more generic bug ( The one I hint to here https://github.com/WordPress/gutenberg/pull/49750#issuecomment-1508353263 ), it's better to stick with string for now until we find a good solution for that one.